File: app/src/main/java/igrek/forceawaken/domain/ringtone/Ringtone.java
package igrek.forceawaken.domain.ringtone;

import android.net.Uri;

import java.io.File;

public class Ringtone {
	
	private File file;
	private String name;
	
	public Ringtone(File file, String name) {
		this.file = file;
		this.name = name;
	}
	
	public File getFile() {
		return file;
	}
	
	public String getName() {
		return name;
	}
	
	public Uri getUri() {
		return Uri.fromFile(file);
	}
	
	@Override
	public String toString() {
		return name;
	}
	
	@Override
	public boolean equals(Object obj) {
		return (obj != null) && (obj instanceof Ringtone) && file.equals(((Ringtone) obj).file);
	}
}

File: app/src/main/java/igrek/forceawaken/ringtone/Ringtone.kt
package igrek.forceawaken.ringtone

import android.net.Uri
import java.io.File

class Ringtone(val file: File, val name: String) {
    val uri: Uri
        get() = Uri.fromFile(file)

    override fun toString(): String {
        return name
    }

    override fun equals(obj: Any?): Boolean {
        return obj != null && obj is Ringtone && file == obj.file
    }
}